tasks/main.yml

```diff
@@ -40,14 +62,46 @@
     - { name: "docker-py", version: "1.10.5" }
     - { name: "docker-compose", version: "1.9.0" }
 
-- name: start docker
-  service: name=docker state=started enabled=yes
+- name: format docker storage
+  filesystem:
+    dev: "{{ docker_storage_device }}"
+    fstype: xfs
+  when: docker_storage_device is defined
   become: True
 
-- name: load overlay kernel module for docker
-  modprobe:
-    name: overlay
-    state: present
+- name: remove default mount
+  mount:
+    path: /mnt
+    src: /dev/vdb
+    state: absent
+  become: True
+  when: docker_storage_device is defined
+
+- name: mount docker storage
+  mount:
+    path: /var/lib/docker
+    src: "{{ docker_storage_device }}"
+    fstype: xfs
+    opts: noatime
+    state: mounted
+  when: docker_storage_device is defined
```

Comment:
What about separating the Docker storage setup into its own playbook that is conditionally included? It's multiple tasks that are all behind the same boolean. Another option would be "block", but a separate playbook could be good for readability.

Reply:
Yes. Maybe a larger refactoring in general is in order.